restapi/src/test/java/io/stargate/web/docsapi/service/CombinedPagingStateTest.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.web.docsapi.service;
+
+import static io.stargate.web.docsapi.service.CombinedPagingState.EXHAUSTED_PAGE_STATE;
+import static io.stargate.web.docsapi.service.CombinedPagingState.deserialize;
+import static io.stargate.web.docsapi.service.CombinedPagingState.serialize;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
+import io.stargate.core.util.ByteBufferUtils;
+import io.stargate.db.PagingPosition.ResumeMode;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+class CombinedPagingStateTest {
+
+  @Nested
+  class IsExhausted {
+
+    @Test
+    void constant() {
+      assertThat(CombinedPagingState.isExhausted(EXHAUSTED_PAGE_STATE)).isTrue();
+    }
+
+    @Test
+    void custom() {
+      assertThat(CombinedPagingState.isExhausted(ByteBuffer.allocate(0))).isTrue();
+      assertThat(CombinedPagingState.isExhausted(ByteBuffer.allocate(1))).isFalse();
+    }
+
+    @Test
+    void nullState() {
+      assertThat(CombinedPagingState.isExhausted(null)).isFalse();
+    }
+  }
+
+  @Nested
+  class MakePagingState {
+
+    @Test
+    void oneNull() {
+      CombinedPagingState combined = new CombinedPagingState(ImmutableList.of(resumeMode -> null));
+      assertThat(combined.makePagingState(ResumeMode.NEXT_PARTITION)).isNull();
+    }
+
+    @Test
+    void allNull() {
+      CombinedPagingState combined =
+          new CombinedPagingState(ImmutableList.of(resumeMode -> null, resumeMode -> null));
+      assertThat(combined.makePagingState(ResumeMode.NEXT_PARTITION)).isNull();
+    }
+
+    @Test
+    void mixed() {
+      CombinedPagingState combined =
+          new CombinedPagingState(
+              ImmutableList.of(resumeMode -> null, resumeMode -> EXHAUSTED_PAGE_STATE));
+      assertThat(combined.makePagingState(ResumeMode.NEXT_PARTITION)).isNotNull();
```

Comment:
is this correct considering the tests above and below?

Reply:
I think it is correct. If one of the results sets was never processed (`null`), while the other  is exhausted, we want a non-null paging state so that the user could ask for more data and cause the result set of the former query to be processed.

Reply:
OK, resolve.